sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/PropertiesAccessCorrectField.cs

```diff
@@ -57,7 +57,9 @@ protected override IEnumerable<FieldData> FindFieldAssignments(IPropertySymbol p
                 FieldData? foundField = null;
                 if (node is AssignmentStatementSyntax assignment && assignment.IsKind(SyntaxKind.SimpleAssignmentStatement))
                 {
-                    foundField = ExtractFieldFromExpression(AccessorKind.Setter, assignment.Left, compilation);
+                    foundField = assignment.Left.DescendantNodesAndSelf().OfType<ExpressionSyntax>()
```

Comment:
The code is similar with the one for C#. Can we extract a few generic methods in order to enable reuse?

Reply:
I didn't like the old version of the rule and considered full refactoring. But there are too many language dependent types => too much work for this sprint. So I went for keeping the original scheme of the rule.
Turning VB into low hanging fruit is a bigger topic.